### PR TITLE
Project Euler #36

### PR DESCRIPTION
**Pull request by ManasGupta1998**
Added #36 in python

### CODE REVIEW

File: code/online_challenges/project_euler/problem_036/problem_036.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    
+def pal(x,b):
+    x=str(x)
+    if x!=x[::-1]:
+        return (False)
+    else:
+        x=int(x)
+        x=base(x,b)
+        x=(str(x))
+        if x!=x[::-1]:
```
Comment (arnavb) [inline code/online_challenges/project_euler/problem_036/problem_036.py]: How about `return x==x[::-1]`

File: code/online_challenges/project_euler/problem_036/problem_036.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            return (False)
+        else:
+            return (True)
+               
+a=int(1000000)
```
Comment (arnavb) [inline code/online_challenges/project_euler/problem_036/problem_036.py]: No need for the int

File: code/online_challenges/project_euler/problem_036/problem_036.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+b=int(2)
```
Comment (arnavb) [inline code/online_challenges/project_euler/problem_036/problem_036.py]: Same here. Also, better variable names would be helpful.

File: code/online_challenges/project_euler/problem_036/problem_036.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+sum=0
+for i in range(1,a+1):
+    if pal(i,b):
```
Comment (arnavb) [inline code/online_challenges/project_euler/problem_036/problem_036.py]: palindrome is a better name

File: code/online_challenges/project_euler/problem_036/problem_036.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        sum+=i
+    else:
```
Comment (arnavb) [inline code/online_challenges/project_euler/problem_036/problem_036.py]: No need for an else statement at all.

File: code/online_challenges/project_euler/problem_036/problem_036.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
+def base(a,y):
+    ans=''
+    while a>0 :
+        ans+=str(a%y)
+        a=a//y   
+    ans=ans[::-1]
+    return (ans)
```
Comment (arnavb) [inline code/online_challenges/project_euler/problem_036/problem_036.py]: Return type doesn't need to be in parentheses for all of your functions.

File: code/online_challenges/project_euler/problem_036/problem_036.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    
+def palindrome(x,base):
+    x=str(x)
+    if x!=x[::-1]:
+        return False
+    else:
+        x=int(x)
+        x=base_check(x,base)
+        x=(str(x))
+        if x!=x[::-1]:
```
Comment (arnavb) [inline code/online_challenges/project_euler/problem_036/problem_036.py]: return x == x[::-1]

File: code/online_challenges/project_euler/problem_036/problem_036.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+            return False
+        else:
+            return True
+               
+number=1000000
+base=2
+sum=0
```
Comment (arnavb) [inline code/online_challenges/project_euler/problem_036/problem_036.py]: Also, sum is a built in name, so don't use it.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**